db/migrate/20210414203512_convert_primary_language_to_iso_code.rb

```diff
@@ -0,0 +1,142 @@
+    end
+    ActiveRecord::Base.transaction do
+      automatic_translations.each do |key, value|
+        # Patient records updated here are solely updated (no History Items included)
+        Patient.where(purged: false).where(primary_language: key).update_all(primary_language: value)
+        Patient.where(purged: false).where(secondary_language: key).update_all(secondary_language: value)
+      end
+      custom_translations.each do |key, value|
+        key = key.to_s
+        if value === 'bilingual english/spanish'
+          Patient.where(purged: false).where(primary_language: key).update_all({ primary_language: 'eng', secondary_language: 'spa' })
+        elsif value === 'bilingual spanish/english'
+          Patient.where(purged: false).where(primary_language: key).update_all({ primary_language: 'spa', secondary_language: 'eng' })
+        elsif TRANSLATION_COMMENTS.include?(key)
+          note = "Primary language was listed as '#{key}' \
+                which could not be matched to the standard language list. This monitoree’s primary \
+                language has been updated to '#{value.nil? ? 'blank' : Languages.all_languages[value.to_sym][:display]}'. You \
+                may update this value."
+          execute <<-SQL.squish
+                INSERT INTO histories (patient_id,created_by,comment,history_type,created_at,updated_at)
+                SELECT id, 'Sara Alert System', "#{note}", 'System Note', CURRENT_TIMESTAMP, CURRENT_TIMESTAMP
+                FROM patients
+                WHERE purged = false AND primary_language = "#{key}"
+          SQL
+          Patient.where(purged: false).where(primary_language: key).update_all(primary_language: value)
+
+          note = "Secondary language was listed as '#{key}' \
+                which could not be matched to the standard language list. This monitoree’s secondary \
+                language has been updated to '#{value.nil? ? 'blank' : Languages.all_languages[value.to_sym][:display]}'. You \
+                may update this value."
+          execute <<-SQL.squish
+                INSERT INTO histories (patient_id,created_by,comment,history_type,created_at,updated_at)
+                SELECT id, 'Sara Alert System', "#{note}", 'System Note', CURRENT_TIMESTAMP, CURRENT_TIMESTAMP
+                FROM patients
+                WHERE purged = false AND secondary_language = "#{key}"
+          SQL
+          Patient.where(purged: false).where(secondary_language: key).update_all(secondary_language: value)
+        end
+      end
+    end
+  end
+
+  def down
+    execute <<-SQL.squish
+        UPDATE patients SET primary_language = legacy_primary_language, secondary_language = legacy_secondary_language
+    SQL
+
+    remove_column :patients, :legacy_primary_language
+    remove_column :patients, :legacy_secondary_language
+
+    History.where(history_type: 'System Note').destroy_all
```

Comment:
I recognize the usefulness of being able to do this - it is a good enough reason to not change the type despite my other comment.

Reply:
On second thought, this might be problematic if we ever used this history type for another purpose, and for whatever reason this migration down was run. It would be safer to add at least some additional qualifier. Example (not tested):
```
History.where(history_type = ? AND comment like ?', 'System Note', 'Primary language was listed as%').destroy_all
```

Reply:
Slight modification for it to work with both `primary_language` and `secondary_language`, but your code works!